src/nvim/api/buffer.c

```diff
@@ -575,6 +576,172 @@ ArrayOf(Dictionary) nvim_buf_get_keymap(Buffer buffer, String mode, Error *err)
   return keymap_array(mode, buf);
 }
 
+/// Like |nvim_set_keymap|, but for a specific buffer.
+///
+/// If not provided in {opts}, |<buffer>| is assumed to be true. If provided
```

Comment:
We don't need the 'buffer' key at all. if `nvim_buf_set_keymap` is used, the mapping is buffer-local, and with `nvim_set_keymap` the mapping is global.